### PR TITLE
br: add errch buf for checkpoint

### LINKED ISSUE #40165: br: full backup in stuck when getting checkpoint
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. br backup full

### 2. What did you expect to see? (Required)
1. no error
### 3. What did you see instead (Required)
1. br get error but in stuck
`[2022/12/23 10:12:17.669 +00:00] [ERROR] [client.go:936] ["can not find leader"] [key=748000000000008EFFB55F698000000000FF0000030000000000FB] [stack="github.com/pingcap/tidb/br/pkg/backup.(*Client).findRegionLeader\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/backup/client.go:936\ngithub.com/pingcap/tidb/br/pkg/backup.(*Client).handleFineGrained\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/backup/client.go:1147\ngithub.com/pingcap/tidb/br/pkg/backup.(*Client).fineGrainedBackup.func2\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/backup/client.go:993"]
[2022/12/23 10:12:17.669 +00:00] [INFO] [client.go:821] ["backup range completed"] [range-sn=13660] [startKey=748000000000008EB55F69800000000000000300] [endKey=748000000000008EB55F698000000000000003FB] [take=7m11.723426318s]
[2022/12/23 10:12:17.669 +00:00] [INFO] [client.go:772] ["Backup Ranges Completed"] [take=1h47m35.629267161s]
[2022/12/23 10:12:17.669 +00:00] [INFO] [backup.go:598] ["wait for flush checkpoint..."]
[2022/12/24 09:55:59.320 +00:00] [WARN] [pd.go:152] ["get timestamp too slow"] ["cost time"=58.430505ms]
[2022/12/24 17:23:43.298 +00:00] [WARN] [pd.go:152] ["get timestamp too slow"] ["cost time"=36.093091ms]
[2022/12/25 14:28:59.297 +00:00] [WARN] [pd.go:152] ["get timestamp too slow"] ["cost time"=34.710956ms]`
### 4. What is your TiDB version? (Required)
/ # /br --version
Release Version: v6.5.0
Git Commit Hash: 4f930817cb78a59ba164a919f063ab7a6d425341
Git Branch: heads/refs/tags/v6.5.0
Go Version: go1.19.3
UTC Build Time: 2022-12-10 14:40:08
Race Enabled: false

### PR DESCRIPTION
**Pull request by Leavrth**
Signed-off-by: Leavrth <jianjun.liao@outlook.com>

### What problem does this PR solve?

Issue Number: close #40165 

Problem Summary:
the sender of error channel is blocked when receiver is closed by other reason.

### What is changed and how it works?
add buffer for the channel
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/checkpoint/checkpoint.go [hunk 1]
```diff
@@ -240,7 +240,7 @@ func StartCheckpointRunnerForTest(ctx context.Context, storage storage.ExternalS
 
 		appendCh: make(chan *CheckpointMessage),
 		metaCh:   make(chan map[string]*RangeGroups),
-		errCh:    make(chan error),
```
Comment (joccau) [inline br/pkg/checkpoint/checkpoint.go:243]: is the count 1 of buffer length enough to deal this issue, if the multiple error is sent to errCh ?
Can we close the errCh before receiver is closed ?

Comment (Leavrth) [reply br/pkg/checkpoint/checkpoint.go:243]: The error incurs only from the flush stage, which is wrapped in a goroutine. Once an error incurs, this goroutine send back the error and exit, so no error will occur again.

File: br/pkg/checkpoint/checkpoint.go [hunk 2]
```diff
@@ -344,6 +344,11 @@ func (r *CheckpointRunner) startCheckpointRunner(ctx context.Context, wg *sync.W
 	return errCh
 }
 
+func (r *CheckpointRunner) sendError(err error) {
+	log.Error("stop checkpoint runner due to error", zap.Error(err))
+	r.errCh <- err
```
Comment (YuJuncen) [inline br/pkg/checkpoint/checkpoint.go:349]: For farther preventing blocking, perhaps we can `select` here:
```go
select {
  case r.errCh <-err:
  default:
    log.Error("errCh blocks", logutil.ShortErr(err))
}

**Review by YuJuncen — APPROVED** [review]

File: br/pkg/checkpoint/checkpoint.go (hunk 2 above)
Comment (Leavrth) [reply br/pkg/checkpoint/checkpoint.go]: done

**Review by joccau — APPROVED** [review]
LGTM

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**